torch/_higher_order_ops/flex_attention.py

```diff
@@ -478,6 +479,20 @@ def flex_attention_fake_tensor_mode(
         return out, logsumexp
 
 
+@flex_attention.py_impl(_CachingTorchDispatchMode)
+def _flex_caching(
+    mode: _CachingTorchDispatchMode, *args: Any, **kwargs: Any
+) -> tuple[torch.Tensor, torch.Tensor]:
+    return flex_attention(*args, **kwargs)
```

Comment:
what makes this correct 🤔

Reply:
The existing mode has some logic to:

(1) check a policy_fn to see if the given op should be recomputed or saved (`mode.policy_fn`) in this case

(2) check if compile is running, and set some FX node metadata to "recompute": https://github.com/pytorch/pytorch/blob/main/torch/utils/checkpoint.py#L1289

(3) potentially stash the output of the op on the mode if we want to save it: https://github.com/pytorch/pytorch/blob/main/torch/utils/checkpoint.py#L1296

Reply:
@drisspg couldn't just be
```
@flex_attention.py_impl(_CachingTorchDispatchMode)
def _flex_caching(
    mode: _CachingTorchDispatchMode, *args: Any, **kwargs: Any
) -> tuple[torch.Tensor, torch.Tensor]:
    return mode.__torch_dispatch__(flex_attention, (), args,kwargs)
```

Then you'd get all the behavior inside the mode that we want

Reply:
I tried this, but hit a snag, might have been a dumb mistake. 

I thought this might just work and get all the above behavior but I might have been confused about how this mode on the stack was behaving

Reply:
@zou3519 I feel like we should be able to make a "default" wrapper here.

Reply:
The plan is to do that but to unblock Titan I am just adding this for now #BE